pkg/controller/daemon/controller.go

```diff
@@ -93,9 +93,15 @@ type DaemonSetsController struct {
 
 	// Daemon sets that need to be synced.
 	queue *workqueue.Type
+
+	// Additional conditions that must be satisfied before running a daemon pod on a node
+	conditions []NodeShouldRunDaemonPod
 }
 
-func NewDaemonSetsController(kubeClient clientset.Interface, resyncPeriod controller.ResyncPeriodFunc) *DaemonSetsController {
+// NodeShouldRunDaemonPod defines a condition that must return true before a daemon pod will be run a node
+type NodeShouldRunDaemonPod func(node *api.Node, ds *extensions.DaemonSet) bool
```

Comment:
This needs to be allowed to return an `error` too.  I could see an argument for just an error, but definitely an error.